doc/docusaurus/docs/working-with-scripts/script-purposes.md

```diff
@@ -0,0 +1,80 @@
+
+## Minting
+
+Minting scripts, sometimes also referred to as minting policies, are used to approve or reject minting of new assets.
+
+An asset on Cardano contains a `CurrencySymbol` and a `TokenName`.
+If a transaction attempts to mint some new assets, then for each unique `CurrencySymbol` in the new assets, the script whose hash matches the `CurrencySymbol` must be included in the transaction, and is executed to validate the minting of that `CurrencySymbol`.
+This is the reason why the `Minting` constructor of `ScriptPurpose` contains a `CurrencySymbol` field.
+
+A minting script should pay attention to the fact that the transaction may attempt to mint multiple assets with the same `CurrencySymbol` but different `TokenNames`.
+In general all these assets with the same `CurrencySymbol` should be checked.
+
+## Rewarding
+
+As previously stated, a UTXO's address can be either a public key address or a script address.
+Both kinds of addresses can optionally have a staking credential.
+A UTXO may contain Ada, and the Ada it contains can be delegated to an SPO to earn staking rewards.
+Staking rewards are deposited into a reward account[^1] corresponding to the staking credential.
+
+A staking credential can contain either a public key hash or a script hash.
+To withdraw rewards from a reward account corresponding to a staking credential that contains a script hash, the script with that particular hash must be included in the transaction, and is executed to validate the withdrawal.
+
+For example, such a script may check that any withdrawal must not take more than 10 Ada at a time, and that any withdrawal must be deposited into one of the allowed addresses.
+
+## Certifying
+
+A certifying script validates one of the following actions a transaction attempts to make: (1) registering a staking credential, and in doing so, creating a reward account associated with the staking credential; (2) de-registering a staking credential, and in doing so, terminating the reward account; (3) delegating a staking credential to a particular delegatee.
+
+In all three cases, if the staking credential in question contains a script hash (as opposed to a public key hash), the script with that hash must be included in the transaction, and is executed to validate the action.
+
+Such a script may, for instance, check that certain signatures be provided for registration, de-registration and delegation, or that the delegatee must be among the allowed delegatees.
+
+## Voting
+
+A voting script validates votes cast by a DRep or a constitutional committee member in a transaction.
+A DRep or a constitution committee member can be associated with a script hash.
+If a transaction contains one or more votes from a DRep or a constitution committee member associated with a script hash, the script with that hash must be included in the transaction, and is executed to approve or reject the vote.
+
```

Comment:
I think it is better to use full name on the first occurrence, and use the abbreviation, acronym or initialism. i.e. 

"A voting script validates votes cast by a Delegated Representative (DRep) or a Constitutional committee member (CCM) in a transaction.

A DRep or a CCM can be associated with a script hash..."